microceph/common/network.go

```diff
@@ -0,0 +1,73 @@
+package common
+
+import (
+	"fmt"
+	"net"
+)
+
+// FindIpOnSubnet scans the host's network interfaces to check if an IP is available
+// for the provided subnet. It returns the found IP address or an empty string.
+func FindIpOnSubnet(subnet string) (string, error) {
+	_, sn, err := net.ParseCIDR(subnet)
+	if err != nil {
+		return "", err
+	}
+
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return "", err
+	}
+
+	for _, iface := range ifaces {
+		addrs, _ := iface.Addrs()
```

Comment:
If there is an issue with the iface we might get back an error here. IMHO it's fine to continue in this case but we might want to log the error